src/Components/WebAssembly/Server/src/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -96,11 +98,12 @@ private static StaticFileOptions CreateStaticFilesOptions(IFileProvider webRootF
                 fileContext.Context.Response.Headers.Append(HeaderNames.CacheControl, "no-cache");
 
```

Comment:
The logic here seems pretty weird. We pick an encoding based on `Accept-Encoding` and our `_supportedEncodings`/`_preferredEncodings`, independently of files on disk, and then serve the matching file if it exists, or uncompressed if not.

Doesn't this mean that, in development where you have `.gz` files but not `.br`, that we're always serving uncompressed content to browsers that accept `br`?

Maybe I'm misreading it, but I would have thought the desired logic would be something like:

 * For each group of encodings in `Accept-Encoding` (ordered by quality, highest first)
    * For the subset of that group that intersects with `_supportedEncoding`, ordered by rank in `_preferredEncoding` (so maybe replace this pair of sets with a single list of supported encodings by our preference)
        * Pick the first one that has a matching file on disk

That would always respect browser accept-encoding quality rankings, combined with our preferences and the availability of precompressed files on disk.